### ISSUE TITLE
Running scene stuck on splash screen - possible scene/script file corruption

### ISSUE THREAD

**Issue by KUNGERMOoN** [godotengine/godot#87441]
### Tested versions

v4.2.1.stable.official [b09f793f5]

### System information

Google Pixel 6, Android 14: Godot v4.2.1.stable - Android - Vulkan (Mobile) - integrated Mali-G78 -  (8 Threads)

### Issue description

I was working on a simple Godot game on my phone, when, the game started to get stuck on the splash screen whenever I run it. I tried restarting Godot and my phone, but it didn't help. I noticed that the issue did not occur when I run any scene that did not use player.tscn (map.tscn, ui.tscn) and that all the exported properties of Player.gd (used in player.tscn) were reset to default values, which might suggest that the scene file or the Player.gd script were corrupted in some way.

### Steps to reproduce

- Download the attached MRP and open it with Godot
(I encountered the issue while using Godot Editor on Android)
- Either run the game or run player.tscn scene or run any other scene that uses player.tscn (ex. main.tscn)
- The game will be stuck on the splash screen
- Notice that when you run any scene other than the mentioned above OR you remove the Player.gd script from the player.tscn OR you run the game from the project list, everything works fine
- Additionally, the exported properties of the Player.gd script were reset to default values in player.tscn

### Minimal reproduction project (MRP)

[MyGame.zip](https://github.com/godotengine/godot/files/14001499/MyGame.zip)

**Comment by AlexOtsuka** [comment]
Hello @KUNGERMOoN,

Trying this on Godot 4.2.1 PlayStore build, and I was stuck on the splash screen just the same. Reassigning the AnimatedSprite2D property in the player scene was all it took for it to run properly, even tho the change isn't immediately reflected in the main scene's inspector. Do your exported properties keep getting reset to their default values?

**Comment by KUNGERMOoN** [comment]
> Do your exported properties keep getting reset to their default values?
No, when I change them and save the node they won't get reset.
I worked around the problem by re-creating the scene and editing & recompiling the script.
Still, it's super weird that the problem occurs when running a project after opening it in the editor, but not when running it from the project list.

**Comment by KUNGERMOoN** [comment]
Testing this on ```Godot v4.2.1.stable.mono - Windows 10.0.19044 - Vulkan (Mobile) - dedicated AMD Radeon HD 8950 (Advanced Micro Devices, Inc.; 27.20.1034.6) - Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 Threads)```, the game starts and gets past the splash screen, but then instantly freezes and stops responding. Also it throws an error:
```E 0:00:01:0047   load_image: Error opening file 'res://icon.svg'.
  <C++ Error>    Condition "f.is_null()" is true. Returning: err
  <C++ Source>   core/io/image_loader.cpp:90 @ load_image()
```
Which is weird considering the icon.svg (which I deleted) doesn't seem to be used anywhere

**Comment by AlexOtsuka** [comment]
> Testing this on `Godot v4.2.1.stable.mono - Windows 10.0.19044 - Vulkan (Mobile) - dedicated AMD Radeon HD 8950 (Advanced Micro Devices, Inc.; 27.20.1034.6) - Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 Threads)`, the game starts and gets past the splash screen, but then instantly freezes and stops responding. Also it throws an error:
> 
> ```
>   <C++ Error>    Condition "f.is_null()" is true. Returning: err
>   <C++ Source>   core/io/image_loader.cpp:90 @ load_image()
> ```
> 
> Which is weird considering the icon.svg (which I deleted) doesn't seem to be used anywhere

I do get that error as well using your MRP in Godot 4.2.1 on Windows, but again, the freeze doesn't happen if I assign the AnimatedSprite2D node in the player scene before running the game so the error seems unrelated.

**Comment by KUNGERMOoN** [comment]
> the freeze doesn't happen if I assign the AnimatedSprite2D node in the player scene before running the game

You're right, that seems to fix the problem. Leaving it unassigned probably caused dereferencing null in the script which might've been what caused the freeze. Is it a correct behaviour? (shouldn't it throw some kind of error rather than freezing the game? - I'm kinda new to GDScript so not sure about that)

> The icon.svg file is used by default on Windows as the game icon in your project list, among other places. You can point to a different icon or empty the text field in Project Settings -> Application -> Config -> Icon so it stops throwing that error.

Oh, it makes more sense now. Thanks!

**Comment by AlexOtsuka** [comment]
I think this is a misunderstanding on your part. As far as I can tell, you've never assigned the value for `sprite` in `Player.gd`:

`@export var sprite: AnimatedSprite2D` only declares the sprite variable and its type, its value here is `null`.

Because your `AnimatedSprite2D` is a node in the tree, it has to be either manually assigned in the inspector or assigned to at runtime like this:

```
func _ready():
	sprite = get_node("AnimatedSprite2D")
```

(more info here: https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-get-node)

The reason your game would only crash inside the inspector is because the release build is more lenient and will avoid crashing as much as possible, as opposed to debug mode inside the editor, which catches errors in order for you to fix them. However, I doubt the animation was actually playing even when you started the game from the project list, right?

As far as I can tell, unless I'm somehow mistaken, there is no bug or file corruption of any kind here.

**Comment by KUNGERMOoN** [comment]
> I think this is a misunderstanding on your part. As far as I can tell, you've never assigned the value for `sprite` in `Player.gd`:
> 
> `@export var sprite: AnimatedSprite2D` only declares the `sprite` variable and its type, its value here is `null`.
> 
> Because your `AnimatedSprite2D` is a node in the tree, it has to be either manually assigned in the inspector or assigned at runtime like this:
> 
> ```
> func _ready():
> 	sprite = get_node("AnimatedSprite2D")
> ```
> 
> (more info here: https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-get-node)
> 
> The reason your game would only crash inside the editor is because the release build is more lenient and will avoid crashing as much as possible, as opposed to debug mode inside the editor, which catches errors in order for you to fix them. However, I doubt the animation was actually playing even when you started the game from the project list, right?
> 
> > It solves the problem, but still, I not sure if not assigning exported property used in the code should cause the game to freeze
> 
> The game crashes not because the property is unassigned, but because you're trying to use that unassigned property at runtime in your code.
> 
> As far as I can tell, unless I'm somehow mistaken, there is no bug or file corruption of any kind here.

You're right, it probably was a mistake on my part, and since you provided evidence it might not be what I originally thought, I was mostly wondering about the unassigned property causing the game to freeze (which you explained in the quoted comment) and the exported properties being reset - which might've been my mistake (especially considering how easy it is to press the wrong button while using the editor on a phone without a mouse).

Anyway, considering it's most likely not an Engine's fault, I'll be closing this issue.

Thank you for your help (and patience) @AlexOtsuka and I hope you have a good day.

**Comment by KUNGERMOoN** [comment]
Update on the topic: it seems like some parts of the project might get non-permanently reset after a game freezes due to the user's code.
For example, after freezing the game and re-opening the editor app, the script editor shows as if some (or sometimes all) script files were empty. This doesn't seem like a script importing problem; I didn't see any errors in the output and even got a code warning regarding the script being empty. 
The problem disappears if you make it so the editor starts with a tab open that is other than the "Script" tab. 
(For example:
- open the editor (the scripts will be seemingly empty)
- navigate to the "2D" tab
- close the editor app without making any changes
- re-open the app and the project
- navigate to the "Scripts" tab
- the scripts will be back)

However, if you try to edit and save the seemingly empty file, it **won't** go back even if you perform the steps I described above (I've lost a bunch of progress that way)

This issue has happened to me ~4 times so far and I believe it might be the cause of the problem I originally described in this issue - before I uploaded the MRP, I remember encountering a freeze and changing some properties of the faulty node (and saving them).

**Comment by KUNGERMOoN** [comment]
Update on the topic: it seems like some parts of the project might get non-permanently reset after a game freezes due to the user's code.
For example, after freezing the game and re-opening the editor app, the script editor shows as if some (or sometimes all) script files were empty. This doesn't seem like a script importing problem; I didn't see any errors in the output and even got a code warning regarding the script being empty. 
The problem disappears if you make it so the editor starts with a tab open that is other than the "Script" tab. 
(For example:
- open the editor (the scripts will be seemingly empty)
- navigate to the "2D" tab
- close the editor app without making any changes
- re-open the app and the project
- navigate to the "Scripts" tab
- the scripts will be back)

However, if you try to edit and save the seemingly empty file, it **won't** go back even if you perform the steps I described above (I've lost a bunch of progress that way)

This issue has happened to me ~4 times so far and I believe it might be the cause of the problem I originally described in this issue - before I uploaded the MRP, I remember encountering a freeze and changing some properties of the faulty node (and saving them).

Edit: This does not seem to be the same problem as #68862 - the provided MRP does not show the scripts as empty, and does provide a parse error that was not present in my issue
